addons/mail/static/src/core/common/thread_model.js

```diff
@@ -526,28 +526,36 @@ export class Thread extends Record {
         return this.isChatChannel ? "@" : "#";
     }
 
-    get lastSelfMessageSeenByEveryone() {
-        const otherMembers = this.channelMembers.filter((member) =>
-            member.persona.notEq(this.store.self)
-        );
-        if (otherMembers.length === 0) {
-            return false;
-        }
-        const otherLastSeenMessageIds = otherMembers
-            .filter((member) => member.seen_message_id)
-            .map((member) => member.seen_message_id.id);
-        if (otherLastSeenMessageIds.length === 0) {
-            return false;
-        }
-        const lastMessageSeenByAllId = Math.min(...otherLastSeenMessageIds);
-        const orderedSelfSeenMessages = this.persistentMessages.filter((message) => {
-            return message.author?.eq(this.store.self) && message.id <= lastMessageSeenByAllId;
-        });
-        if (!orderedSelfSeenMessages || orderedSelfSeenMessages.length === 0) {
-            return false;
-        }
-        return orderedSelfSeenMessages.slice().pop();
-    }
+    lastSelfMessageSeenByEveryone = Record.one("Message", {
+        compute() {
+            const otherMembers = this.channelMembers.filter((member) =>
+                member.persona.notEq(this.store.self)
+            );
+            if (otherMembers.length === 0) {
+                return false;
+            }
+            const otherLastSeenMessageIds = otherMembers
+                .filter((member) => member.seen_message_id)
+                .map((member) => member.seen_message_id.id);
+            if (otherLastSeenMessageIds.length === 0) {
+                return false;
+            }
+            const lastMessageSeenByAllId = Math.min(...otherLastSeenMessageIds);
+            let res;
+            for (let i = this.persistentMessages.length - 1; i >= 0; i--) {
+                const message = this.persistentMessages[i];
+                if (!message.author?.eq(this.store.self)) {
```

Comment:
May be worth caching "all persistent messages of current user" to avoid recomputing this field whenever a new message from someone else is received?